snowpack/src/types/snowpack.ts

```diff
@@ -47,7 +48,9 @@ export interface PluginRunOptions {
 }
 
 /** map of extensions -> code (e.g. { ".js": "[code]", ".css": "[code]" }) */
-export type PluginLoadResult = string | SnowpackBuildMap;
+export type PluginLoadResult = SnowpackBuildMap;
```

Comment:
I shifted the `string` part of the type to the method definition for consistency with `PluginTransformResult`.  (Including the `string` in that type makes casting a lot less useful.)

Reply:
Thanks, but we actually don’t want that as part of the `SnowpackBuildMap`. We do allow polymorphism when it comes to interfacing with plugins for flexibility and a better plugin DX. But for our own internal types we want them to be very strict. Moving that type up to `SnowpackBuildMap` means we have to add many more typechecks in our app than we would otherwise.

So basically we want plugin APIs to be loose and polymorphic; internal types to be strict.

Reply:
I pushed the `string` down to `load()`, not up to `SnowpackBuildMap`.  It's basically a no-op (just one other spot in the code had to be changed) and `SnowpackBuildMap` hasn't changed one bit.  It just looked weird to have `PluginLoadResult` include `string` and `PluginTransformResult` not.

Reply:
Ah understood. Sorry for misunderstanding